src/Map/src/MapFactory.php

```diff
@@ -0,0 +1,186 @@
+        if (\array_key_exists('extra', $marker)) {
+            if (!\is_array($marker['extra'])) {
+                throw new \InvalidArgumentException(\sprintf('The "extra" parameter must be an array, "%s" given.', get_debug_type($marker['extra'])));
+            }
+            $extra = $marker['extra'];
+            unset($marker['extra']);
+        }
+
+        if (\count($marker) > 0) {
+            throw new \InvalidArgumentException(\sprintf('Unknown marker parameters: "%s".', implode('", "', array_keys($marker))));
+        }
+
+        return new Marker($point, $title, $infoWindow, $extra);
+    }
+
+    /**
+     * @param array{content?: string, headerContent?: string} $infoWindow
+     */
+    private static function createInfoWindow(array $infoWindow): InfoWindow
+    {
+        $headerContent = null;
+        if (\array_key_exists('headerContent', $infoWindow)) {
+            if (!\is_string($infoWindow['headerContent'])) {
+                throw new \InvalidArgumentException(\sprintf('The "header" parameter must be a string, "%s" given.', get_debug_type($infoWindow['headerContent'])));
+            }
+
+            $headerContent = $infoWindow['headerContent'];
+            unset($infoWindow['headerContent']);
+        }
+
+        $content = null;
+        if (\array_key_exists('content', $infoWindow)) {
+            if (!\is_string($infoWindow['content'])) {
+                throw new \InvalidArgumentException(\sprintf('The "content" parameter must be a string, "%s" given.', get_debug_type($infoWindow['content'])));
+            }
+
+            $content = $infoWindow['content'];
+            unset($infoWindow['content']);
+        }
+
+        if (\count($infoWindow) > 0) {
+            throw new \InvalidArgumentException(\sprintf('Unknown "infoWindow" parameters: "%s".', implode('", "', array_keys($infoWindow))));
+        }
+
+        if (!$headerContent && !$content) {
+            throw new \InvalidArgumentException('The "infoWindow" parameter must have at least one of "header" or "content" values.');
+        }
+
+        return new InfoWindow($headerContent, $content);
+    }
+
+    /**
+     * @param array{lat?: float, lng?: float} $point
+     */
+    private static function createPoint(array $point): Point
+    {
+        if (!isset($point['lat']) || !isset($point['lng'])) {
+            throw new \InvalidArgumentException('The Point parameter must be an array with "lat" and "lng" keys.');
+        }
+        if (!is_numeric($point['lat']) || !is_numeric($point['lng'])) {
+            throw new \InvalidArgumentException('The "lat" and "lng" values must be numeric.');
+        }
+
+        $lat = (float) $point['lat'];
+        $lng = (float) $point['lng'];
+        unset($point['lat'], $point['lng']);
+
+        if (\count($point) > 2) {
+            throw new \InvalidArgumentException(\sprintf('Unknown Point parameters: "%s".', implode('", "', array_keys($point))));
+        }
+
+        return new Point($lat, $lng);
```

Comment:
Given my previous comments, I imagine something like this (exceptions message can be rewritten): 
```suggestion
        return new Point(
            (float) $point['lat'] ?? throw new \InvalidArgumentException('Parameter "lat" is missing in "Point" definition.'), 
            (float) $point['lng'] ?? throw new \InvalidArgumentException('Parameter "lng" is missing in "Point" definition.')
        );
```